cfn.yaml

```diff
@@ -0,0 +1,136 @@
+      - PROD
+    Default: CODE
+  DeployBucket:
+    Description: Bucket where RiffRaff uploads artifacts on deploy
+    Type: String
+    Default: membership-dist
+Resources:
+  ExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action: sts:AssumeRole
+      Path: /
+      Policies:
+        - PolicyName: logs
+          PolicyDocument:
+            Statement:
+              Effect: Allow
+              Action:
+                - logs:CreateLogGroup
+                - logs:CreateLogStream
+                - logs:PutLogEvents
+              Resource: arn:aws:logs:*:*:*
+        - PolicyName: lambda
+          PolicyDocument:
+            Statement:
+              Effect: Allow
+              Action:
+                - lambda:InvokeFunction
+              Resource: "*"
+        - PolicyName: PrivateBucket
+          PolicyDocument:
+            Statement:
+               Effect: Allow
+               Action: s3:GetObject
+               Resource: !Sub arn:aws:s3:::membership-private/${Stage}/*
+        - PolicyName: SQSPutInQueue
+          PolicyDocument:
+            Statement:
+              Effect: Allow
+              Action: "*"
+              Resource: !Sub arn:aws:sqs:${AWS::Region}:${AWS::AccountId}:salesforce-outbound-messages-${Stage}
+  SalesforceMessageHandlerLambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      FunctionName: !Sub salesforce-message-handler-${Stage}
+      Code:
+        S3Bucket:
+          Ref: DeployBucket
+        S3Key: !Sub ${Stack}/${Stage}/${App}/${App}.jar
+      Environment:
+        Variables:
+          Stage: !Ref Stage
+          Stack: !Ref Stack
+          App: !Ref App
+      Description: parse salesforce outbound messages and put them in a queue
+      Handler: com.gu.salesfoce.messageHandler.Lambda::handleRequest
+      MemorySize: 512
+      Role: !GetAtt ExecutionRole.Arn
+      Runtime: java8
+      Timeout: 300
+
+  MessageHandlerApiPermission:
+          Type: AWS::Lambda::Permission
+          Properties:
+              Action: lambda:invokeFunction
+              FunctionName: !Sub salesforce-message-handler-${Stage}
+              Principal: apigateway.amazonaws.com
+          DependsOn: SalesforceMessageHandlerLambda
+
+  MessageHandlerApi:
+      Type: "AWS::ApiGateway::RestApi"
+      Properties:
+          Description: Api to handle salesforce outbound messages
+          Name: !Sub salesforce-message-handler-${Stage}
+
+  MessageHandlerApiResource:
+      Type: AWS::ApiGateway::Resource
+      Properties:
+          RestApiId: !Ref MessageHandlerApi
+          ParentId: !GetAtt [MessageHandlerApi, RootResourceId]
+          PathPart: contact
+      DependsOn: MessageHandlerApi
+
+  ContactMethod:
+      Type: AWS::ApiGateway::Method
+      Properties:
+          AuthorizationType: NONE
```

Comment:
no, they provide a more complicated solution using client certificates but it's not directly supported by AWS API gateway.